.github/workflows/build.yml

```diff
@@ -0,0 +1,96 @@
+            chmod a+x /bin/repo
+          fi
+
+      - name: Sync mirror
+        run: |
+          # Create/maintain a mirror of all manifest repositories in runner
+          # persistent storage, and use that as reference when initializing the
+          # build repo, effectively reducing the time spent fetching git repos to
+          # almost nothing.
+          #
+          # If for some reason, the repo mirror becomes corrupted, a pipeline with
+          # $REBUILD_REPO_MIRROR set to a non-empty value will remove the entire
+          # mirror, and rebuild it from scratch.
+          #
+          # Note the use of fd 9 to hold a lock while accessing the $REPO_MIRROR
+          # path, so we can handle parallel jobs.  First we open the lock file on fd
+          # 9, then we lock it, and close/unlock it after repo sync.
+          if [ -n "$CI_ONIRO_RUNNER_PERSISTENT_STORAGE" ] ; then
+            REPO_MIRROR="$CI_ONIRO_RUNNER_PERSISTENT_STORAGE/repo-mirrors" ;
+            mkdir -p $(dirname $REPO_MIRROR) ;
+            exec 9>$REPO_MIRROR.lock ; flock 9 ;
+            if [ -n "$REBUILD_REPO_MIRROR" ] ; then
+              rm -rf "$REPO_MIRROR" ;
+            fi ;
+            if [ ! -e "$REPO_MIRROR" ] ; then
+              echo "Creating new repo mirror @ $REPO_MIRROR" ;
+              mkdir -p "$REPO_MIRROR" ;
+              pushd "$REPO_MIRROR" ;
+              repo init -u "$CI_ONIRO_MANIFEST_URL" -m "$CI_ONIRO_MANIFEST_NAME" -b "$CI_ONIRO_MANIFEST_BRANCH" --mirror ;
+            else
+              echo "Reusing repo mirror @ $REPO_MIRROR" ;
+              pushd "$REPO_MIRROR" ;
+              repo init -u "$CI_ONIRO_MANIFEST_URL" -m "$CI_ONIRO_MANIFEST_NAME" -b "$CI_ONIRO_MANIFEST_BRANCH";
+            fi ;
+            repo sync --no-clone-bundle ;
+            repo forall -c 'git lfs pull'
+            popd ;
+          fi
+
+      - name: Fetch sources
+        run: |
+          # Create the build environment in a repo subdir
+          mkdir repo && cd repo
+          repo init \
+            $(test -n "${CI_ONIRO_RUNNER_PERSISTENT_STORAGE:-}" && echo --reference "$CI_ONIRO_RUNNER_PERSISTENT_STORAGE"/repo-mirrors) \
+            --manifest-url "$CI_ONIRO_MANIFEST_URL" \
+            --manifest-name "$CI_ONIRO_MANIFEST_NAME" \
+            --manifest-branch "$CI_ONIRO_MANIFEST_BRANCH"
+          repo sync --no-clone-bundle
+          repo forall -c 'git lfs pull'
+          # Release repo mirror lock if held
+          if [ -n "$REPO_MIRROR_ARG" ] ; then exec 9>&- ; fi
+
+      - name: Download prebuilts
+        run: |
+          mkdir -p $CI_ONIRO_RUNNER_PERSISTENT_STORAGE/OpenHarmony_canary_prebuilts
+          ln -s $CI_ONIRO_RUNNER_PERSISTENT_STORAGE/OpenHarmony_canary_prebuilts OpenHarmony_canary_prebuilts
+          cd repo
+          ./build/prebuilts_download.sh
+
+      - name: Build
+        run: |
+          cd repo
+          ./build.sh --product-name rk3568
```

Comment:
For ccache you would need --ccache as an option here. In the docker build the cache directory would be /root/.ccache by default, which is not part of the volume shared with the host (like the sources). If we want to use ccache this would need to be redirected to /home/openharmony.

Reply:
I enabled shared ccache which is stored in persistent storage